File: tests/fixtures/sample_project_java/src/main/java/com/example/billing/Invoices.java
package com.example.billing;

import com.example.users.User;

/** A billing invoice for a user. */
class Invoice {
    public long userId;
    public long amountCents;
}

/** Invoice generation and delivery, built on the Stripe client. */
public class Invoices {
    /** Charge a user's invoice through Stripe and return the charge id. */
    public static String chargeInvoice(StripeClient client, Invoice invoice) {
        return client.chargeCustomer(invoice.amountCents);
    }

    /** Render an invoice and hand it to the email transport. */
    public static boolean sendInvoiceEmail(User user, Invoice invoice) {
        return user.id == invoice.userId;
    }
}

File: tests/fixtures/sample_project_csharp/src/Billing/Invoices.cs
using Acme.Users;

namespace Acme.Billing;

/// <summary>A billing invoice for a user.</summary>
public class Invoice
{
    public long UserId;
    public long AmountCents;
}

/// <summary>Invoice generation and delivery, built on the Stripe client.</summary>
public class Invoices
{
    /// <summary>Charge a user's invoice through Stripe and return the charge id.</summary>
    public static string ChargeInvoice(StripeClient client, Invoice invoice)
    {
        return client.ChargeCustomer(invoice.AmountCents);
    }

    /// <summary>Render an invoice and hand it to the email transport.</summary>
    public static bool SendInvoiceEmail(User user, Invoice invoice)
    {
        return user.Id == invoice.UserId;
    }
}
